File: dodam-application/dodam-rest-api/src/main/java/b1nd/dodam/restapi/support/meal/MealController.java
package b1nd.dodam.restapi.support.meal;

import b1nd.dodam.neis.meal.client.NeisMealClient;
import b1nd.dodam.neis.meal.client.data.Meal;
import b1nd.dodam.restapi.support.data.Response;
import b1nd.dodam.restapi.support.data.ResponseData;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping(value = "/meal")
@RequiredArgsConstructor
public class MealController {

    private final NeisMealClient neisMealClient;
    private final MealUseCase mealUseCase;

    @GetMapping
    public ResponseData<Meal> getMeal(@RequestParam int year,
                                      @RequestParam int month,
                                      @RequestParam int day) {
        return ResponseData.ok("급식 조회 성공", neisMealClient.getMeal(year, month, day));
    }

    @GetMapping("/month")
    public ResponseData<List<Meal>> getMealOfMonth(@RequestParam int year,
                                                   @RequestParam int month) {
        return ResponseData.ok("한달 급식 조회 성공", neisMealClient.getMealOfMonth(year, month));
    }

    @PostMapping
    public Response registerMeal(@RequestBody @Valid Meal meal) {
        mealUseCase.saveMeal(meal);
        return Response.ok("급식 등록 성공");
    }

}

File: services/service-neis/src/main/kotlin/com/b1nd/dodamdodam/neis/presentation/meal/MealController.kt
package com.b1nd.dodamdodam.neis.presentation.meal

import com.b1nd.dodamdodam.core.security.annotation.authentication.UserAccess
import com.b1nd.dodamdodam.core.security.passport.enumerations.RoleType
import com.b1nd.dodamdodam.neis.application.meal.MealUseCase
import org.springframework.format.annotation.DateTimeFormat
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDate
import java.time.YearMonth

@RestController
@RequestMapping("/meal")
class MealController(
    private val mealUseCase: MealUseCase,
) {
    @UserAccess
    @GetMapping
    fun getMealsByDate(@RequestParam date: LocalDate) =
        mealUseCase.getMealsByDate(date)

    @UserAccess(roles = [RoleType.ADMIN])
    @PostMapping("/sync")
    fun syncMeals(@RequestParam @DateTimeFormat(pattern = "yyyy-MM") yearMonth: YearMonth) =
        mealUseCase.syncMeals(yearMonth)
}